src/main/kotlin/org/neo4j/graphql/AugmentationHandler.kt

```diff
@@ -0,0 +1,22 @@
+package org.neo4j.graphql
+
+import graphql.schema.*
+
+abstract class AugmentationHandler(val schemaConfig: SchemaConfig) {
+    companion object {
+        const val QUERY = "Query"
+        const val MUTATION = "Mutation"
+    }
+
+    abstract fun augmentType(type: GraphQLFieldsContainer, buildingEnv: BuildingEnv)
```

Comment:
does this have to be abstract?

Reply:
since this class is the blueprint of a Handler: yes

Reply:
if it was not abstract it could be handled by a closure